src/line.rs

```diff
@@ -71,17 +73,28 @@ impl EditLine {
         let mut start_pos = range.start_offset().expect("offset") as usize;
         let mut end_pos = range.end_offset().expect("offset") as usize;
 
-        if self.text.starts_with(Self::COSMETIC_SPACE) {
+        // Remove the cosmetic space on a new line UNLESS we're simply
+        // making another new line.
+        if self.text.starts_with(Self::COSMETIC_SPACE)
```

Comment:
Maybe this should be a common accessor for the line (self.logical_text().is_empty() or something)?

Reply:
I created a `logical_text_is_empty` -> currently it qualifies as empty OR cosmetic space -- should it just be a check for cosmetic space?

Reply:
I think we should try to be rigid about this because otherwise I'm worried there will be too many cases to handle.

We could try to be clear that:

- The "logical text" can never have a cosmetic space in it.
- The "physical text" can never be empty.

And then we wouldn't have to check for both conditions (empty OR cosmetic space) in either the logical or physical text.